### DISCUSSION TITLE
No harware SPI on NUCLEO-F746ZG board?

### DISCUSSION THREAD

**Discussion by mdaeron** [micropython/micropython#13734] [STM32 / Pyboard]
I'm experimenting with a NUCLEO-F746ZG board. Based on the pinout I expected there to be up to six hardware SPI peripherals. But `machine.SPI(0)` yields the error "ValueError: SPI(0) doesn't exist". Same with `machine.SPI(1)` as well as `pyb.SPI()`.

Is hardware SPI not implemented on this board or am I missing something?

**Comment by davefes** [comment]
I would check the mpconfigboard.h file to see if the SPI is used:
```
// SPI buses
#define MICROPY_HW_SPI3_NSS         (pin_A4)
#define MICROPY_HW_SPI3_SCK         (pin_B3)
#define MICROPY_HW_SPI3_MISO        (pin_B4)
#define MICROPY_HW_SPI3_MOSI        (pin_B5)
```

**Comment by mdaeron** [comment]
Understood. So only SPI(3) is defined there, ans I assume I could add SPI(0) or SPI if I wanted to by editing `mpconfigboard.h`. Thanks!

**Reply by davefes** [reply]
Maybe, then you have to build the firmware yourself.  I'd check SPI3 first.

**Reply by mdaeron** [reply]
Great, thanks again.
